File: common/src/main/java/me/wyne/wutils/common/inventory/IUtils.java
package me.wyne.wutils.common.inventory;

import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

public final class IUtils {

    public static boolean addItem(Inventory inventory, ItemStack... items) {
        return inventory.addItem(items).isEmpty();
    }

    public static boolean addItem(Player player, ItemStack... items) {
        return addItem(player.getInventory(), items);
    }

    public static void addOrDrop(Player player, ItemStack... items) {
        var exceed = player.getInventory().addItem(items);
        exceed.values().forEach(item -> player.getLocation().getWorld().dropItem(player.getLocation(), item));
    }

}

File: commonKt/src/main/kotlin/me/wyne/wutils/common/kotlin/inventory/IUtils.kt
package me.wyne.wutils.common.kotlin.inventory

import me.wyne.wutils.common.inventory.IUtils
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack

fun Inventory.addItem(vararg items: ItemStack) =
    IUtils.addItem(this, *items)

fun Player.addItem(vararg items: ItemStack) =
    IUtils.addItem(this, *items)

fun Player.addOrDrop(vararg items: ItemStack) =
    IUtils.addOrDrop(this, *items)